Test backendtests

StefanEvanghelides opened the pull request:
Preparing the pull request  early. Before merging, every backend class must be tested. We should easily achieve 80-90% coverage level in this package.

StefanEvanghelides committed:
Initial commit. Tried to mock NetworkHandler inside device class

StefanEvanghelides committed:
Device class is fully tested now

StefanEvanghelides committed:
included the IOException in DeviceTest

StefanEvanghelides committed:
Tested RequiredInfo class

StefanEvanghelides committed:
added tests for ComFaultException; added setters as well

StefanEvanghelides committed:
added packageTest too all backendTest classes.

StefanEvanghelides committed:
alter: added packageTest to all backend test classes

StefanEvanghelides committed:
prepared tests for ServerCollectionsInteractor class

StefanEvanghelides committed:
updated ServerCollectionsInteractor's description

StefanEvanghelides committed:
Complete the tests throwing IOExceptions

Next are: tests finishing successfully and tests throwing a
ComFaultException

FrankTN commented on Hestia/app/src/androidTest/java/com/rugged/application/hestia/backend/NetworkHandlerTest.java:
```diff
@@ -39,12 +48,12 @@ public void setAndGetPortTest() {
+        Integer responseCodeSuccess = 200;
+        Integer responseCodeFail = 404;
+        Integer responseCodeNull = null;
+        assertTrue(dummyHandler.isSuccessfulRequest(responseCodeSuccess));
```
Nice!

FrankTN reviewed (changes requested):
Nice refactoring of some earlier classes and good to see some new tests. Unfortunately, some tests still expect ComFaultExceptions. This needs to be changed otherwise it will not build

StefanEvanghelides committed:
added more tests for ServerCollectionsInteractor

StefanEvanghelides committed:
Finished testing ServerCollectionsInteractor

StefanEvanghelides commented:
TODO's:
- Improve Activator tests
- Tests for all deserializers.
- Check coverage levels for each class, see how I can improve them

UPDATE: Activator and ActivatorState tests improved
UPDATE 2: Updated tests for deserializers
UPDATE 3: Coverage level will increase automatically if we optimize the equals() and hashCode() methods

StefanEvanghelides committed:
added tests for Activator

StefanEvanghelides committed:
finished testing Activator class

StefanEvanghelides committed:
refined ActivatorState tests. added more

StefanEvanghelides committed:
changed the name from DeserializerTest to ActivatorDeserializerTest; updated tests here

StefanEvanghelides committed:
deleted commented out code

StefanEvanghelides committed:
small tweaks; changed Travis avd's API level to 19 (from 22)

People reportedly fixed a bug inside Travis by modifying the API level
to 19.

See:
https://stackoverflow.com/questions/39457305/android-testing-waited-for-the-root-of-the-view-hierarchy-to-have-window-focus

LarsHoldijk commented on Hestia/app/src/androidTest/java/com/rugged/application/hestia/backend/NetworkHandlerTest.java:
```diff
@@ -64,23 +73,23 @@ public void getDefaultPathTest() {
+        NetworkHandler handlerDifferentProperties = new NetworkHandler(IP, port);
+        assertNotNull(handlerDifferentProperties);
+        assertFalse(dummyHandler.equals(handlerDifferentProperties));
     }
```
This could  be divided into multiple methods, this makes it easier to see which assert fails.

LarsHoldijk commented on Hestia/app/src/androidTest/java/com/rugged/application/hestia/backend/models/ActivatorStateTest.java:
```diff
@@ -1,66 +1,143 @@
+        JsonPrimitive stringPrimitive = new JsonPrimitive(DEFAULT_STRING_RAW_STATE);
+        assertNotNull(stringPrimitive);
+        assertEquals(stringPrimitive, stringActivatorState.getRawStateJSON());
+    }
```
Instead of indication the different sections it would be better to divide them evenly.

LarsHoldijk reviewed (changes requested):
Really impressive work Stefan. I just had some minor things that have to do with the structure of the tests. To make it more readable and easier to see what failed we could divide the tests. Some have at the moment different sections where different scenarios are tested.

Btw, mocking the network handler with mockito looks really good. Nice find!

StefanEvanghelides committed:
Divided ActivatorState and NetworkHandler methods evenly
